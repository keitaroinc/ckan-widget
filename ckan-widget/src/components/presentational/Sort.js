import React, { Component } from 'react';

class Sort extends Component {
  render() {
    return (
      <div className="input-group">
        <div className="input-group-prepend">
            <label htmlFor="datasets_per_page" className="input-group-text">Order by:</label>
        </div>
        <select className="custom-select" value={this.props.sort} onChange={e => this.props.handleSort(e.target.value)}>
          <option value="score desc, metadata_modified desc">Relevance</option>
          <option value="title_string asc">Name Ascending</option>
          <option value="title_string desc">Name Descending</option>
          <option value="metadata_modified desc">Last Modified</option>
          <option value="views_recent desc">Popular</option>
        </select>
      </div>
    );
  }
}

export default Sort;
