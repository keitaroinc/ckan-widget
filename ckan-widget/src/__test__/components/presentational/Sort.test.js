import React from 'react';
import { shallow } from 'enzyme';
import Sort from '../../../components/presentational/Sort';

let component;
let handleSort

describe('Sort', () => {
  beforeEach(() => {
    handleSort = jest.fn();
    component = shallow(
      <Sort handleSort={handleSort} sort={'score desc, metadata_modified desc'} />
    );
  })

  it('should be defined', () => {
    expect(Sort).toBeDefined();
  });

  it('should render correctly', () => {
    expect(component.instance().props.sort).toBe('score desc, metadata_modified desc')
    expect(component).toMatchSnapshot();
  });

  it('should handle change correctly', () => {
    const selectElement = component.find('select')
    selectElement.simulate('change', {target: {value: 25}})
    expect(handleSort.mock.calls.length).toEqual(1);
  })
});
