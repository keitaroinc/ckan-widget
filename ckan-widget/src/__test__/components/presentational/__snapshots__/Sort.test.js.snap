// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sort
    handleSort={[Function]}
    sort="score desc, metadata_modified desc"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="input-group"
>
        <div
                className="input-group-prepend"
        >
                <label
                        className="input-group-text"
                        htmlFor="order_by"
                        title="Order by"
                >
                        <i
                                className="material-icons"
                        >
                                sort
                        </i>
                </label>
        </div>
        <select
                className="custom-select"
                id="order_by"
                onChange={[Function]}
                value="score desc, metadata_modified desc"
        >
                <option
                        value="score desc, metadata_modified desc"
                >
                        Relevance
                </option>
                <option
                        value="title_string asc"
                >
                        Name Ascending
                </option>
                <option
                        value="title_string desc"
                >
                        Name Descending
                </option>
                <option
                        value="metadata_modified desc"
                >
                        Last Modified
                </option>
                <option
                        value="views_recent desc"
                >
                        Popular
                </option>
        </select>
</div>,
      "className": "mr-0",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="input-group-prepend"
>
            <label
                        className="input-group-text"
                        htmlFor="order_by"
                        title="Order by"
            >
                        <i
                                    className="material-icons"
                        >
                                    sort
                        </i>
            </label>
</div>,
          <select
            className="custom-select"
            id="order_by"
            onChange={[Function]}
            value="score desc, metadata_modified desc"
>
            <option
                        value="score desc, metadata_modified desc"
            >
                        Relevance
            </option>
            <option
                        value="title_string asc"
            >
                        Name Ascending
            </option>
            <option
                        value="title_string desc"
            >
                        Name Descending
            </option>
            <option
                        value="metadata_modified desc"
            >
                        Last Modified
            </option>
            <option
                        value="views_recent desc"
            >
                        Popular
            </option>
</select>,
        ],
        "className": "input-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <label
              className="input-group-text"
              htmlFor="order_by"
              title="Order by"
>
              <i
                            className="material-icons"
              >
                            sort
              </i>
</label>,
            "className": "input-group-prepend",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="material-icons"
>
                sort
</i>,
              "className": "input-group-text",
              "htmlFor": "order_by",
              "title": "Order by",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "sort",
                "className": "material-icons",
              },
              "ref": null,
              "rendered": "sort",
              "type": "i",
            },
            "type": "label",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value="score desc, metadata_modified desc"
>
                Relevance
</option>,
              <option
                value="title_string asc"
>
                Name Ascending
</option>,
              <option
                value="title_string desc"
>
                Name Descending
</option>,
              <option
                value="metadata_modified desc"
>
                Last Modified
</option>,
              <option
                value="views_recent desc"
>
                Popular
</option>,
            ],
            "className": "custom-select",
            "id": "order_by",
            "onChange": [Function],
            "value": "score desc, metadata_modified desc",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Relevance",
                "value": "score desc, metadata_modified desc",
              },
              "ref": null,
              "rendered": "Relevance",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Name Ascending",
                "value": "title_string asc",
              },
              "ref": null,
              "rendered": "Name Ascending",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Name Descending",
                "value": "title_string desc",
              },
              "ref": null,
              "rendered": "Name Descending",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Last Modified",
                "value": "metadata_modified desc",
              },
              "ref": null,
              "rendered": "Last Modified",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Popular",
                "value": "views_recent desc",
              },
              "ref": null,
              "rendered": "Popular",
              "type": "option",
            },
          ],
          "type": "select",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="input-group"
>
          <div
                    className="input-group-prepend"
          >
                    <label
                              className="input-group-text"
                              htmlFor="order_by"
                              title="Order by"
                    >
                              <i
                                        className="material-icons"
                              >
                                        sort
                              </i>
                    </label>
          </div>
          <select
                    className="custom-select"
                    id="order_by"
                    onChange={[Function]}
                    value="score desc, metadata_modified desc"
          >
                    <option
                              value="score desc, metadata_modified desc"
                    >
                              Relevance
                    </option>
                    <option
                              value="title_string asc"
                    >
                              Name Ascending
                    </option>
                    <option
                              value="title_string desc"
                    >
                              Name Descending
                    </option>
                    <option
                              value="metadata_modified desc"
                    >
                              Last Modified
                    </option>
                    <option
                              value="views_recent desc"
                    >
                              Popular
                    </option>
          </select>
</div>,
        "className": "mr-0",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="input-group-prepend"
>
              <label
                            className="input-group-text"
                            htmlFor="order_by"
                            title="Order by"
              >
                            <i
                                          className="material-icons"
                            >
                                          sort
                            </i>
              </label>
</div>,
            <select
              className="custom-select"
              id="order_by"
              onChange={[Function]}
              value="score desc, metadata_modified desc"
>
              <option
                            value="score desc, metadata_modified desc"
              >
                            Relevance
              </option>
              <option
                            value="title_string asc"
              >
                            Name Ascending
              </option>
              <option
                            value="title_string desc"
              >
                            Name Descending
              </option>
              <option
                            value="metadata_modified desc"
              >
                            Last Modified
              </option>
              <option
                            value="views_recent desc"
              >
                            Popular
              </option>
</select>,
          ],
          "className": "input-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <label
                className="input-group-text"
                htmlFor="order_by"
                title="Order by"
>
                <i
                                className="material-icons"
                >
                                sort
                </i>
</label>,
              "className": "input-group-prepend",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="material-icons"
>
                  sort
</i>,
                "className": "input-group-text",
                "htmlFor": "order_by",
                "title": "Order by",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "sort",
                  "className": "material-icons",
                },
                "ref": null,
                "rendered": "sort",
                "type": "i",
              },
              "type": "label",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value="score desc, metadata_modified desc"
>
                  Relevance
</option>,
                <option
                  value="title_string asc"
>
                  Name Ascending
</option>,
                <option
                  value="title_string desc"
>
                  Name Descending
</option>,
                <option
                  value="metadata_modified desc"
>
                  Last Modified
</option>,
                <option
                  value="views_recent desc"
>
                  Popular
</option>,
              ],
              "className": "custom-select",
              "id": "order_by",
              "onChange": [Function],
              "value": "score desc, metadata_modified desc",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Relevance",
                  "value": "score desc, metadata_modified desc",
                },
                "ref": null,
                "rendered": "Relevance",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Name Ascending",
                  "value": "title_string asc",
                },
                "ref": null,
                "rendered": "Name Ascending",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Name Descending",
                  "value": "title_string desc",
                },
                "ref": null,
                "rendered": "Name Descending",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Last Modified",
                  "value": "metadata_modified desc",
                },
                "ref": null,
                "rendered": "Last Modified",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Popular",
                  "value": "views_recent desc",
                },
                "ref": null,
                "rendered": "Popular",
                "type": "option",
              },
            ],
            "type": "select",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
